File: 5lesson/HW/view/Menu.java

import java.util.Scanner;

import 5lesson.HW.controller.UserController;
import 5lesson.HW.model.User;

public class Menu {
	private static boolean play = true;
	private static String name;
	private static String surname;
	private static String patronymic;
	private static float balance;

	public static void showMenu() {
		UserController userController = new UserController();
		int menuCase = 0;
		Scanner scanner = new Scanner(System.in);
		int i;
		while (play) {
			System.out.println("1. Добавить клиента.\n" +
					"2. Удалить клиента.\n" +
					"3. Изменить данные клиента.\n" +
					"4. Показать список клиентов.\n" +
					"5. Увеличить баланс клиента.\n" +
					"6. Уменьшить баланс клиента.\n" +
					"7. Выход.");
			menuCase = scanner.nextInt();
			switch (menuCase) {
				case 1:
					System.out.print("Введите имя клиента: ");
					scanner.nextLine();
					name = scanner.nextLine();
					System.out.print("Введите фамилию клиента: ");
					surname = scanner.nextLine();
					System.out.print("Введите отчество клиента: ");
					patronymic = scanner.nextLine();
					userController.saveUser(name, surname, patronymic);
					break;
				case 2:
					i = 0;
					for (User user : userController.getAllUsers()) {
						System.out.println(i + " " + user);
						i++;
					}
					System.out.print("Выберите идентификатор для удаления: ");
					i = scanner.nextInt();
					try {
						userController.deleteUser(userController.getAllUsers().get(i));
					} catch (IndexOutOfBoundsException e) {
						System.out.println("Такого пользователя не существует!");
					}
					i = 0;
					for (User user : userController.getAllUsers()) {
						System.out.println(i + " " + user);
						i++;
					}
					System.out.println();
					break;
				case 3:
					i = 0;
					for (User user : userController.getAllUsers()) {
						System.out.println(i + " " + user);
						i++;
					}
					System.out.print("Выберите идентификатор для изменения: ");
					// Scanner scanner = new Scanner(System.in);
					i = scanner.nextInt();
					System.out.print("Введите имя клиента: ");
					scanner.nextLine();
					name = scanner.nextLine();
					System.out.print("Введите фамилию клиента: ");
					surname = scanner.nextLine();
					System.out.print("Введите отчество клиента: ");
					patronymic = scanner.nextLine();
					// upd.close();
					try {
						balance = userController.getAllUsers().get(i).getBalance();
						User updatedUser = new User(name, surname, patronymic);
						updatedUser.setBalance(balance);
						userController.updateUser(i, updatedUser);
					} catch (IndexOutOfBoundsException e) {
						System.out.println("Такого пользователя не существует!");
					}
					i = 0;
					for (User user : userController.getAllUsers()) {
						System.out.println(i + " " + user);
						i++;
					}
					System.out.println();
					break;
				case 4:
					i = 0;
					for (User user : userController.getAllUsers()) {
						System.out.println(i + " " + user);
						i++;
					}
					System.out.println();
					break;
				case 5:
					i = 0;
					for (User user : userController.getAllUsers()) {
						System.out.println(i + " " + user);
						i++;
					}
					System.out.print("Выберите идентификатор для увеличения баланса: ");
					// Scanner scanner = new Scanner(System.in);
					i = scanner.nextInt();
					System.out.println(
							"Текущий баланс клиента = " + userController.getAllUsers().get(i).getBalance());
					System.out.print("На сколько увеличить баланс клиента: ");
					try {
						userController.addFunds(i, scanner.nextInt());
					} catch (IndexOutOfBoundsException e) {
						System.out.println("Такого пользователя не существует!");
					}

					i = 0;
					for (User user : userController.getAllUsers()) {
						System.out.println(i + " " + user);
						i++;
					}
					System.out.println();
					break;
				case 6:
					i = 0;
					for (User user : userController.getAllUsers()) {
						System.out.println(i + " " + user);
						i++;
					}
					System.out.print("Выберите идентификатор для уменьшения баланса: ");
					i = scanner.nextInt();
					System.out.println(
							"Текущий баланс клиента = " + userController.getAllUsers().get(i).getBalance());
					System.out.print("На сколько уменьшить баланс клиента: ");
					try {
						userController.removeFunds(i, scanner.nextInt());
					} catch (IndexOutOfBoundsException e) {
						System.out.println("Такого пользователя не существует!");
					}
					i = 0;
					for (User user : userController.getAllUsers()) {
						System.out.println(i + " " + user);
						i++;
					}
					System.out.println();
					break;
				case 7:
					play = false;
					scanner.close();
					break;
				default:
					System.out.println("Выбран неверный пункт меню!");
					break;
			}

		}

	}
}

File: Seminar7/DZ/py/Menu.py
from UserController import UserController
from User import User


class Menu:
    __play = True

    def showList(userController):
        i = 0
        for user in userController.getAllUsers():
            print(f"{i} {user}")
            i += 1
        print()

    def showMenu(self):
        userController = UserController()
        menuCase = 0
        while (self.__play):
            print("1. Добавить клиента.",
                  "2. Удалить клиента.",
                  "3. Изменить данные клиента.",
                  "4. Показать список клиентов.",
                  "5. Увеличить баланс клиента.",
                  "6. Уменьшить баланс клиента.",
                  "7. Выход.", sep='\n')
            menuCase = int(input())
            match menuCase:
                case 1:
                    name = input("Введите имя клиента: ")
                    surname = input("Введите фамилию клиента: ")
                    patronymic = input("Введите отчество клиента: ")
                    userController.saveUser(name, surname, patronymic)
                case 2:
                    Menu.showList(userController)
                    i = int(input("Выберите идентификатор для удаления: "))
                    try:
                        userController.deleteUser(
                            userController.getAllUsers()[i])
                    except:
                        print("Такого пользователя не существует!")
                    Menu.showList(userController)
                case 3:
                    Menu.showList(userController)
                    i = int(input("Выберите идентификатор для изменения: "))
                    name = input("Введите имя клиента: ")
                    surname = input("Введите фамилию клиента: ")
                    patronymic = input("Введите отчество клиента: ")
                    try:
                        balance = userController.getAllUsers()[i].getBalance()
                        updatedUser = User(name, surname, patronymic)
                        updatedUser.setBalance(balance)
                        userController.updateUser(i, updatedUser)
                    except:
                        print("Такого пользователя не существует!")
                    Menu.showList(userController)
                case 4:
                    Menu.showList(userController)
                case 5:
                    Menu.showList(userController)
                    i = int(
                        input("Выберите идентификатор для увеличения баланса: "))
                    print(
                        f"Текущий баланс клиента = {userController.getAllUsers()[i].getBalance()}")
                    try:
                        userController.addFunds(
                            i, int(input("На сколько увеличить баланс клиента: ")))
                    except:
                        print("Такого пользователя не существует!")

                    Menu.showList(userController)
                case 6:
                    Menu.showList(userController)
                    i = int(
                        input("Выберите идентификатор для уменьшения баланса: "))
                    print(
                        f"Текущий баланс клиента = {userController.getAllUsers()[i].getBalance()}")
                    try:
                        userController.removeFunds(
                            i, int(input("На сколько уменьшить баланс клиента: ")))
                    except:
                        print("Такого пользователя не существует!")
                    Menu.showList(userController)
                case 7:
                    self.__play = False
                case _:
                    System.out.println("Выбран неверный пункт меню!")
